src/NServiceBus.Core/Sagas/ISagaIdGenerator.cs

```diff
@@ -0,0 +1,17 @@
+namespace NServiceBus.Sagas
+{
+    using System;
+
+    /// <summary>
+    /// Saga id generator.
+    /// </summary>
+    public interface ISagaIdGenerator
+    {
+        /// <summary>
+        /// Generates a saga id based on property name and property value.
```

Comment:
I just realized that property name and value is optional (if a custom finder is used) so how about:

> Generates a custom saga id

?